app/views/admin/settings/index.html.erb

```diff
@@ -5,77 +5,87 @@
 <!--main content-->
 <%= form_for @admin_setting, :url => admin_setting_path(@admin_setting), html: { method: "put", class: "manage" } do |f| %>
   <%= error_messages_for @admin_setting %>
-  <fieldset>
-    <legend><%= ts("Accounts and Invitations") %></legend>
-    <h3 class="landmark heading"><%= ts("Accounts and Invitations") %></h3>
-    <dl>
-
-      <dt><%= f.check_box :account_creation_enabled %></dt>
-      <dd><%= f.label :account_creation_enabled, ts("Account creation enabled") %></dd>
-      <dt><%= f.check_box :creation_requires_invite %></dt>
-      <dd><%= f.label :creation_requires_invite, ts("Account creation requires invitation") %></dd>
-      <dt><%= f.check_box :request_invite_enabled %></dt>
-      <dd><%= f.label :request_invite_enabled, ts("Users can request invitations") %></dd>
-
-      <dt><%= f.check_box :invite_from_queue_enabled %></dt>
-      <dd><%= f.label :invite_from_queue_enabled, ts("Invite from queue enabled (People can add themselves to the queue and invitations are sent out automatically)") %></dd>
-
-      <dt><%= f.label :invite_from_queue_number, ts("Number of people to invite from the queue at once") %></dt>
-      <dd><%= f.text_field :invite_from_queue_number, :size => "3" %></dd>
 
-      <dt><%= f.label :invite_from_queue_frequency, ts("How often (in days) should we invite people from the queue") %></dt>
-      <dd><%= f.text_field :invite_from_queue_frequency, :size => "3" %></dd>
-
-      <dt><%= f.label :days_to_purge_unactivated, ts("How many weeks you have to activate your account before we purge it") %></dt>
-      <dd><%= f.text_field :days_to_purge_unactivated, :size => "3" %></dd>
-    </dl>
-  </fieldset>
-
-  <fieldset>
-    <legend><%= ts("Turn Off Support Form") %></legend>
-    <h3 class="landmark heading"><%= ts("Turn Off Support Form") %></h3>
-    <dl>
-      <dt><%= f.check_box :disable_support_form %></dt>
-      <dd>
-        <%= f.label :disable_support_form, ts("Turn off support form") %>
-        <legend><%= f.label :disabled_support_form_text, ts("Disabled support form text") %></legend>
-        <p><%= allowed_html_instructions %></p>
-        <%= f.text_area :disabled_support_form_text %>
-      </dd>
-    </dl>
-  </fieldset>
-  
+  <% if AdminSettingPolicy.user_has_correct_role?(current_admin, nil) %>
```

Comment:
Admins with any of the settings roles (superadmin, support, policy_and_abuse, or tag_wrangling) should be able to see all settings, even if they can update some of them.

We can disable the inputs of the settings they cannot update, instead of hiding them entirely.